app/javascript/react/components/SessionFilters/MobileSessionFilters.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useMemo } from "react";
+
+import { useTranslation } from "react-i18next";
+import { selectFixedSessionsState } from "../../store/fixedSessionsSelectors";
+import { useAppSelector } from "../../store/hooks";
+import { selectMobileSessionsState } from "../../store/mobileSessionsSelectors";
+import { selectSelectedSessionType } from "../../store/sessionFiltersSlice";
+import { SessionTypes } from "../../types/filters";
+import { CloseButton } from "../Map/Legend/Legend.style";
+import * as S from "./SessionFilters.style";
+import { SessionTypeToggle } from "./SessionTypeToggle";
+
+interface MobileSessionFiltersProps {
+  onClose: () => void;
+}
+
+const MobileSessionFilters = ({ onClose }: MobileSessionFiltersProps) => {
+  const { t } = useTranslation();
+
+  const selectedSessionType = useAppSelector(selectSelectedSessionType);
+  const fixedSessionsState = useAppSelector(selectFixedSessionsState);
+  const mobileSessionsState = useAppSelector(selectMobileSessionsState);
+
+  let sessionsCount;
+
+  useMemo(() => {
+    if (selectedSessionType === SessionTypes.FIXED) {
+      sessionsCount = fixedSessionsState.sessions.length;
+    } else if (selectedSessionType === SessionTypes.MOBILE) {
+      sessionsCount = mobileSessionsState.sessions.length;
+    }
+  }, [fixedSessionsState, mobileSessionsState, selectedSessionType]);
```

Comment:
to utilize `useMemo` optimization it would be better to use `useMemo` to directly set the _sessionsCount_. 

I'd consider using a `switch` statement instead of `if/else` - while both are valid, switch handles memory allocation and memory address usage more efficiently. I have no strong feelings about that, but I think it might be better in the Aircasting case handling lots of data.

Reply:
Thank you!